cpp/src/strings/count_matches.cu

```diff
@@ -62,11 +63,14 @@ struct count_dispatch_fn {
 
   template <int stack_size>
   std::unique_ptr<column> operator()(column_device_view const& d_strings,
+                                     size_type output_size,
                                      rmm::cuda_stream_view stream,
                                      rmm::mr::device_memory_resource* mr)
   {
+    CUDF_EXPECTS(output_size >= d_strings.size(), "Unexpected output size");
+
     auto results = make_numeric_column(
-      data_type{type_id::INT32}, d_strings.size() + 1, mask_state::UNALLOCATED, stream, mr);
+      data_type{type_id::INT32}, output_size, mask_state::UNALLOCATED, stream, mr);
```

Comment:
It seems a little awkward to me that the output size is the parameter here. The algorithm doesn't care about the size of the column, it's just an element-wise transform. While it would require a bit more duplication, it would be a cleaner division if the calling code was responsible for making the column and then passed a `mutable_column_view` (or device view) to `count_matches`.

Reply:
If this were not a detail API I would probably agree with you. I would rather do it this way than duplicate creating the column everywhere before calling it.